nixos/modules/programs/zsh/zsh.nix

```diff
@@ -67,9 +67,7 @@ in
       };
 
       promptInit = mkOption {
-        default = ''
-          autoload -U promptinit && promptinit && prompt walters
```

Comment:
You're right. My problem is with how oh-my-zsh works. It changes the content of `interactiveShellInit`, which is executed *before* `promptInit`. As such, the default value will always have precedence. If I change the order and executes it *after*, another problem arises: any change I make to `promptInit` will be overriden by `oh-my-zsh`.
I compromised by putting the current default value before `interactiveShellInit` and the actual `promptInit` after.

You are right that this will always cause `walters` to be loaded, which may be undesirable.
Another more sensible solution would be to change the default value in case oh-my-zsh is used. Would you rather have that?